File: University.Service/University.Dto/Zystem/ZVARDto.cs
using System;
using System.Data;
using System.Collections.Generic;
using System.Runtime.Serialization;

using University.Dto.Base;

namespace University.Dto.Zystem
{
    [DataContract]
    public class ZVARDto
    {
        #region General Property

        [DataMember]
        public string ZRCONO { get; set; }

        [DataMember]
        public string ZRBRNO { get; set; }

        [DataMember]
        public string ZRVANO { get; set; }

        [DataMember]
        public string ZRVANA { get; set; }

        [DataMember]
        public string ZRVATY { get; set; }

        [DataMember]
        public string ZRVAVL { get; set; }

        [DataMember]
        public string ZRVASQ { get; set; }

        [DataMember]
        public string ZRREMA { get; set; }

        [DataMember]
        public string ZRSYST { get; set; }

        [DataMember]
        public string ZRSTAT { get; set; }

        [DataMember]
        public decimal ZRRCST { get; set; }

        [DataMember]
        public decimal ZRCRDT { get; set; }

        [DataMember]
        public decimal ZRCRTM { get; set; }

        [DataMember]
        public string ZRCRUS { get; set; }

        [DataMember]
        public decimal ZRCHDT { get; set; }

        [DataMember]
        public decimal ZRCHTM { get; set; }

        [DataMember]
        public string ZRCHUS { get; set; }

        #endregion
    }

    public class ZVARDtoMap : Mapper<ZVARDto>
    {
        protected override ZVARDto PopulateItem(IDataRecord dr)
        {
            ZVARDto dto = new ZVARDto();
            for (int i = 0; i < dr.FieldCount; i++)
            {
                MapProperty(dto, dr.GetName(i), dr[i]);
            }
            return dto;
        }
    }
}

File: University/lib/Dto/Zystem/ZVARDto.dart

class ZVARDto
{
  String ZRCONO = "";
  String ZRBRNO = "";
  String ZRVANO = "";
  String ZRVANA = "";
  String ZRVATY = "";
  String ZRVAVL = "";
  String ZRVASQ = "";
  String ZRREMA = "";
  String ZRSYST = "";
  String ZRSTAT = "";
  double ZRRCST = 0;
  double ZRCRDT = 0;
  double ZRCRTM = 0;
  String ZRCRUS = "";
  double ZRCHDT = 0;
  double ZRCHTM = 0;
  String ZRCHUS = "";

  ZVARDto({
    this.ZRCONO = "",
    this.ZRBRNO = "",
    this.ZRVANO = "",
    this.ZRVANA = "",
    this.ZRVATY = "",
    this.ZRVAVL = "",
    this.ZRVASQ = "",
    this.ZRREMA = "",
    this.ZRSYST = "",
    this.ZRSTAT = "",
    this.ZRRCST = 0,
    this.ZRCRDT = 0,
    this.ZRCRTM = 0,
    this.ZRCRUS = "",
    this.ZRCHDT = 0,
    this.ZRCHTM = 0,
    this.ZRCHUS = "",
  });

  factory ZVARDto.fromJson(Map<String, dynamic> json) {
    return ZVARDto(
      ZRCONO: json["ZRCONO"] ?? "",
      ZRBRNO: json["ZRBRNO"] ?? "",
      ZRVANO: json["ZRVANO"] ?? "",
      ZRVANA: json["ZRVANA"] ?? "",
      ZRVATY: json["ZRVATY"] ?? "",
      ZRVAVL: json["ZRVAVL"] ?? "",
      ZRVASQ: json["ZRVASQ"] ?? "",
      ZRREMA: json["ZRREMA"] ?? "",
      ZRSYST: json["ZRSYST"] ?? "",
      ZRSTAT: json["ZRSTAT"] ?? "",
      ZRRCST: json["ZRRCST"] ?? 0,
      ZRCRDT: json["ZRCRDT"] ?? 0,
      ZRCRTM: json["ZRCRTM"] ?? 0,
      ZRCRUS: json["ZRCRUS"] ?? "",
      ZRCHDT: json["ZRCHDT"] ?? 0,
      ZRCHTM: json["ZRCHTM"] ?? 0,
      ZRCHUS: json["ZRCHUS"] ?? "",
    );
  }

  Map toMap()
  {
    var map = <String, dynamic>{};
    map["ZRCONO"] = ZRCONO;
    map["ZRBRNO"] = ZRBRNO;
    map["ZRVANO"] = ZRVANO;
    map["ZRVANA"] = ZRVANA;
    map["ZRVATY"] = ZRVATY;
    map["ZRVAVL"] = ZRVAVL;
    map["ZRVASQ"] = ZRVASQ;
    map["ZRREMA"] = ZRREMA;
    map["ZRSYST"] = ZRSYST;
    map["ZRSTAT"] = ZRSTAT;
    map["ZRRCST"] = ZRRCST;
    map["ZRCRDT"] = ZRCRDT;
    map["ZRCRTM"] = ZRCRTM;
    map["ZRCRUS"] = ZRCRUS;
    map["ZRCHDT"] = ZRCHDT;
    map["ZRCHTM"] = ZRCHTM;
    map["ZRCHUS"] = ZRCHUS;
    return map;
  }
}
